### ISSUE TITLE
[Console Scheduler] allow to specify the command name used for mutexName

### ISSUE THREAD

**Issue by VinceG** [laravel/ideas#1388]
Hey,

Right now the onOneServer does not work if the command run has a variable in it. For example

`$schedule->exec('node ' . base_path('scripts/script.js'))->hourly()->onOneServer();`

the above code runs on each of the servers, here is why:

since we use automatic deployments the base_path changes for example we have 3 servers running so the base path on each varies

on server No. 1 /var/www/site/release/120
on server No. 2 /var/www/site/release/121
on server No. 3 /var/www/site/release/122

Especially with auto scaling when it scales up and down the releases path is usually different between servers.

This creates a problem with onOneServer since the mutexName it uses to create a unique name for the task uses the command as part of the hash.

return 'framework'.DIRECTORY_SEPARATOR.'schedule-'.sha1($this->expression.$this->command);

i'd like to know if you'll be open for a PR that allows to specify a name for a task so we can have something like this:

`return 'framework'.DIRECTORY_SEPARATOR.'schedule-'.sha1( $this->name ?: $this->expression.$this->command );`

then specify the task as

`$schedule->exec('node ' . base_path('scripts/script.js'))->hourly()->name('script.js')->onOneServer();`

this will solve this issue and will have no regression.

i can submit a PR for this if this is something you'll consider to merge. 

Thank You.

**Comment by paras-malhotra** [comment]
@VinceG you can use `preventOverlapsUsing` for setting a custom mutex.

**Comment by VinceG** [comment]
@paras-malhotra thank you! didn't see that. this requires creating a custom mutex class to be used? the mutexName() method in the create and exists methods is still being pulled from the schedule event itself.

**Comment by paras-malhotra** [comment]
Yes, it would require creating a custom event mutex class and overriding the methods to implement your logic for mutex name.

**Comment by VinceG** [comment]
@paras-malhotra Thank You. I'd still like to know if the ability to specify a name for the mutexName method is something that they will consider, since that'll be easier. essentially the class i'll create will be identical to the CacheSchedulingMutex and instead of $event->mutexName() i'll just have to replace it with a method within the class that returns something that'll be unique but take into account the path that might be different across servers.

**Comment by VinceG** [comment]
@paras-malhotra From past experience i think you are right, these type of ideas usually get closed quickly, worth a shot though.

To be honest this is not a specific use case, the release path number gets out of sync in auto scaling since in AWS the auto scaling is done from an AMI, that AMI might be old and the release number most likely going to be lower than the current servers. Or maybe this is just how we have it working, i am not sure.

Regardless, worst case i'll just add that class and change the method.

Thank You!

**Comment by michaeldyrynda** [comment]
Instead of an incrementing release number, could you not use something that better and more consistently identifies the release? Like the release tag or commit sha? That would at least give consistency across a fleet of scaling servers AND solve the issue under discussion.

**Comment by VinceG** [comment]
@michaeldyrynda Thank again. the release_name is not documented so i wasn't aware of it.

it went from

```
➤ Executing task config:current
+--------------+---------+
| Host         | Current |
+--------------+---------+
| 10.10.10.223 | 247     |
| 10.10.11.130 | 225     |
| 10.10.12.10  | 246     |
+--------------+---------+
✔ Ok
```

to 

```
➤ Executing task config:current
+--------------+---------------+
| Host         | Current       |
+--------------+---------------+
| 10.10.10.223 | 2018_11_14_22 |
| 10.10.11.130 | 2018_11_14_22 |
| 10.10.12.10  | 2018_11_14_22 |
+--------------+---------------+
✔ Ok
```

**Comment by michaeldyrynda** [comment]
@VinceG what is the release_name? Is that part of Envoyer, or something else? I'm not sure what the context is :)

**Comment by VinceG** [comment]
@michaeldyrynda Sorry, we use Deployer for our deployments currently, by default the release paths are numeric increments of the current release. Documentation does not specify you can use an option to set the release name (folder name) as an option.

What i did was just add the following to our deployment script.

```
set('release_name', function(){
	return date('Y_m_d_H');
});
```

that way it's always the same for all servers.

**Comment by michaeldyrynda** [comment]
Awesome! Definitely a better solution if you define a consistent release name for sure. If the numbers differ, it makes it harder to identify what’s current on each host.
